### ISSUE TITLE
Current status and subsequent development plan

### ISSUE THREAD

**Issue by adriatic** [aurelia-auth0/new-samples#5]
While the plan should be obvious - just fix the **[initial app](https://github.com/auth0-community/auth0-aurelia-samples/tree/master/01-Login)** and add the other, missing ones, I want to explain a few more details.

1. I propose to augment the "standard" set of five quick-start samples set by creating each of them as either ES6 or Typescript version (please check the article **[Aurelia Auth0 samples - structure and content](https://github.com/aurelia-auth0/new-samples/issues/3)** for details of this augmented set.

2. In addition to these (10 + 2 basic) samples, I propose to add 2 more "real life" samples that illustrate how to use Auth0 services in the more elaborate context. Details to be discussed **[here](https://github.com/aurelia-auth0/new-samples/issues/4)**.

3. Once all of the samples are finished and tested, the whole **[new-samples](https://github.com/aurelia-auth0/new-samples)** collection will be "packaged" as one large PR for the **[original collection](https://github.com/auth0-community/auth0-aurelia-samples)**

**Comment by aaguiarz** [comment]
Looks good

About the structure, in most Quickstarts we don't have the 00-Basic step, but it's something we keep discussing if it makes sense or not. For angular/react we are thinking of starting from a scaffolded app like the one created by 'create react app', or 'ng new'. Is there any something similar Aurelia?

Other details:

- We usually call 'auth0-variables.js' the file you are calling 'auth0-config.js', for consistency I'd use the first one.

- We have an integration tests that automates 'Login' for SPAs, and to be able to run it in any SPA sample we need the buttons to have the same ID, you can see the IDs we use in the test itself https://github.com/auth0-samples/spa-quickstarts-tests/blob/master/lock_login_test.js.

- What are you using _ignoreLocalStorage_ for?

Thanks

**Comment by adriatic** [comment]
> Is there any something similar Aurelia?

Yes, as described here: https://aurelia.io/docs/cli (at some point everyone reverted back to console (no GUI) based desktop application generators, including Aurelia). Since Aurelia CLI has a rich repertoire of types of applications it can generate, I picked the application type, closest one to the **[existing one](https://github.com/auth0-community/auth0-aurelia-samples/tree/master/01-Login)** written by Chenkie.
The **['Project configuration' section](https://github.com/aurelia-auth0/new-samples/tree/master/esnext-webpack/00-Basic#building-the-app)** of the every README.md file (for each 00-Basic app) shows the generated application's structure:

![image](https://user-images.githubusercontent.com/2712405/40260768-7887655e-5aca-11e8-9291-398b8f7f1787.png)

> We usually call 'auth0-variables.js' the file you are calling 'auth0-config.js', for consistency I'd use the first one.

I will comply with all standards you point me to - this choice was inherited from Ryan Chenkie, I believe

> We have an integration tests that automates 'Login' for SPAs,

Thanks for letting me know, I will follow this rule. At the moment, all of my apps are derived from the respected basic version, which have Aurelia standard testing code. 

> What are you using `ignoreLocalStorage` for?

Temporary flag, which I used to ignore the case where I could be authenticated because the JWT token is still alive. Not something that will survive my "development stage"

**Comment by adriatic** [comment]
**On the first paragraph in [this section above](https://github.com/aurelia-auth0/new-samples/issues/5#issuecomment-390351160)**
___The description of building the initial code (00-basic) will be provided in the README.md file for 01-Login application and the 00-Basic goes away.___

**On the [login test for SPA](https://github.com/auth0-samples/spa-quickstarts-tests/blob/master/lock_login_test.js)**
___Will follow this rule___
